eclipse-theia/theia
# Set initial size and position of secondary windows

#### What it does
This PR implements the initial size and positions of secondary windows. It adds a preference called `Secondary Window Placement` allowing the user to choose the placement of the secondary windows between:
- `originalSize` (default): same size as the original widget.
- `halfSize`: half the size of the running Theia application
- `fullSize`: the same size as the running Theia application.

In addition, another preference (`alwaysOnTop`) is introduced that allows the user to decide if they want the secondary window to always stay above all other windows, including those of different applications.


#### How to test



1.  Open Theia electron or browser application.
2.  Install the `drawio` extension.
3.  Modify the `Secondary Window Placement` preference.
4.  Create a new `test.drawio` file or create a new terminal and open it as a secondary window.
5.  The secondary window' size and placement should match the chosen preference.
6.  Repeat steps 3-5 with the other preferences.

#### Follow-ups

The secondary window popup might not line up perfectly with the original widget since the API used to get the window dimensions has inconsistencies across different browsers and operating systems.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
